plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -942,7 +1049,8 @@ def main():
             hostpci=dict(type='dict'),
             hotplug=dict(type='str'),
             hugepages=dict(choices=['any', '2', '1024']),
-            ide=dict(type='dict'),
+            ide=dict(type='dict', default=None),
```

Comment:
```suggestion
            ide=dict(type='dict'),
```

Reply:
@aminvakil did you just want this one removed, or all of the new `default=None`?

Reply:
Actually I think it's better not have any defaults in module and rely on proxmox defaults, but I was not sure and didn't have time to go through the new code added and check what would happen if the defaults are removed.

I think deleted defaults from #850 are should be removed and don't be re-added again.

I suppose it's better for new options to not have any defaults too, but again it must be checked in code to prevent any failures without the defaults, as far as I know proxmox has default for all cloud-init variables, but I haven't checked api yet.

Reply:
In that case the code is essentially ok. No defaults are re-added, and the only new defaults are `None`, which is the default value for default anyway. It's still cleaner to remove them. @morph027 could you please remove the remaining `default=None`s for the new options? (I spot at least two for existing options, let's just keep them in this PR.)

Reply:
Sure, you're right, I didn't remember they were set to `None`.

Reply:
Jip, will do.